index.js

```diff
@@ -0,0 +1,95 @@
+"use strict";
+
+const cookieSession = require("cookie-session");
+const express = require("express");
+const expressHandlebars = require("express-handlebars");
+const githubAuth = require("connect-oauth-github");
+const github = require("github-request");
+const postRequest = require("request").post;
+const Slack = require("./lib/slack");
+const version = require("./package").version;
+const config = require("./config");
+
+const slackClient = new Slack(config.slack);
+
+const app = express();
+app.use(cookieSession({
+	name: "session",
+	keys: config.web.cookieKeys
+}));
+
+app.engine("handlebars", expressHandlebars({ defaultLayout: "main" }));
+app.set("view engine", "handlebars");
+
+const gha = githubAuth.createClient({
+	id: config.github.clientId,
+	secret: config.github.secret,
+	scope: "user:email"
+});
+
+app.get("/", (request, response) => {
+	response.render("index");
+});
+
+app.get("/auth", gha.handshake);
+
+app.get("/signup", gha.authorize, (request, response) => {
+	let ghaUser = gha.users[request.sessionID];
+
+	github.request({
+		path: "/user?access_token=" + ghaUser.accessToken,
+	}, (error, data) => {
+		if (error) {
+			// TODO: Implement error page
+			return response.status(500).render("error");
+		}
+
+		// TODO: Filter on data.login blacklist
+
+		let ageInDays = (Date.now() - new Date(data.created_at)) / (1000 * 60 * 60 * 24);
+		if (ageInDays < config.github.minimumAge) {
+			// TODO: Check whitelist for approved accounts
```

Comment:
This was based on a request made by you ;-)

> The main reason I want to tie to GitHub is because it gives us a more holistic ability to moderate. In addition to banning from slack, we can also ban on GitHub
I do also think that we should add a requirement that the GitHub account they use to sign in is more than 30 days old
That would pretty effectively prevent trolls from using throwaway accounts
And we can also have a process for people to request exceptions by emailing us

Though re-reading that now, I realize I may have been making an assumption that you meant building the exceptions into the invitation tool. We can just as easily manually invite them. I'm sure it would be the same amount of time for someone to do the manual invitation as it would to add the user to a whitelist, so let's just go with manual invitations. We can make the page tell the user to send us an email and we can generate a link that would even populate their email for them.